elastic/kibana
# [ILM] Removed freeze action from Cold phase

Fixes https://github.com/elastic/kibana/issues/103335 

## Summary

- Removed the freeze action from ILM (cold phase)
- added a clean up to the form serializer to remove freeze action from any policies that still specify it
- deleted freeze from the Console autocomplete.

### How to test
1. Create an ILM policy with freeze action via Dev Tools
```
PUT _ilm/policy/test
{
  "policy": {
    "phases": {
      "hot": {
        "min_age": "0ms",
        "actions": {
          "rollover": {
            "max_age": "30d",
            "max_primary_shard_size": "50gb"
          }
        }
      },
      "cold": {
        "min_age": "30d",
        "actions": {
          "freeze": {}
        }
      }
    }
  }
}
```

2. Check that the freeze action is not displayed in the cold phase
3. Check that the freeze action is not displayed in the request flyout
4. After saving the policy in the UI, check that it doesn't have freeze action defined anymore 
```
GET _ilm/policy/test
```

## Release Note
Since ILM freeze action is a no-op, it was also removed from Index Lifecycle Management.
